homeassistant/components/generic/config_flow.py

```diff
@@ -309,6 +316,40 @@ def register_preview(hass: HomeAssistant) -> None:
     hass.data[DOMAIN][IMAGE_PREVIEWS_ACTIVE] = True
 
 
+async def register_stream_preview(hass: HomeAssistant, config) -> str:
+    """Set up preview for camera stream during config flow."""
+    hass.data.setdefault("camera", {})
+
+    # Need to load the camera prefs early to avoid errors generating the stream
+    # if the user does not already have the stream component loaded.
+    if hass.data.get(DATA_CAMERA_PREFS) is None:
```

Comment:
Makes sense to do this, and perhaps sharing with the init code to avoid any drift would be useful.

Reply:
Thanks @allenporter for the review! 

Was this what you had in mind?: 86802a166

Will look at the other points asap (work quite busy at the moment)

Reply:
Yeah!